### DISCUSSION TITLE
Cypress automatically adds Cookie to a backend call when preforming click action on a button

### DISCUSSION THREAD

**Discussion by SoundararajanMaersk** [cypress-io/cypress#27197] [Questions and Help]
Cookie is automatically adding as request header to backend api call when performing a click action on a button on our application page. We tried with intercepting the call and clearing/deleting cookies but not working. **We want to remove the cookie getting added as a request header.**

cy.clearCookie(CookieName);
Cypress.automation('clear:cookie', { name: 'CookieName' })
delete req.headers["Cookie"]

Can anyone one help on this ?

**Comment by portyu9** [comment]
Hi @SoundararajanMaersk 

The `Cookie` header is normally added by the **browser from its cookie jar**. It is not something your button/click handler necessarily added itself.

If the request should genuinely be unauthenticated, the preferred fix is to remove the relevant cookie **before** triggering the request:

```js
cy.clearCookie('session_id', {
  domain: 'api.example.com',
})

cy.get('[data-cy=button]').click()
```

Pay attention to the domain. By default `cy.clearCookie()` uses the hostname of the current page, so if the request goes to another host/subdomain you may have been clearing the wrong cookie.

If your goal is specifically to alter one intercepted request, use the lowercase header key:

```js
cy.intercept('POST', '**/your-endpoint', (req) => {
  delete req.headers.cookie
}).as('request')

cy.get('[data-cy=button]').click()

cy.wait('@request')
  .its('request.headers')
  .should('not.have.property', 'cookie')
```

`req.headers["Cookie"]` and `req.headers.cookie` are not equivalent JavaScript object-property lookups even though HTTP header names themselves are case-insensitive.

I would also avoid `Cypress.automation('clear:cookie', ...)`; that is an internal Cypress API. `cy.clearCookie()` / `cy.clearCookies()` are the supported public APIs.

If the cookie reappears after clearing it, inspect what happens between `clearCookie()` and the click—the application or another response may simply be setting it again.

The full discussion was reviewed; there were no screenshots/attachments or replies.

Please mark this answer as accepted if it helped, thank you.
